dot/network/notifications.go

```diff
@@ -247,6 +242,22 @@ func (s *Service) broadcastExcluding(info *notificationsProtocol, excluding peer
 			logger.Trace("sending handshake", "protocol", info.protocolID, "peer", peer, "message", hs)
 			err = s.host.send(peer, info.protocolID, hs)
 		} else {
+			if s.host.messageCache != nil {
+				var encMsg []byte
+				encMsg, err = msg.Encode()
+				if err != nil {
+					logger.Error("failed to encode message", "peer", peer, "error", err)
+					continue
+				}
+
+				var added bool
+				added, err = s.host.messageCache.Put(peer, encMsg)
```

Comment:
it might be nice to update `Put` to take `Message`, and encode the message inside `Put` so it doesn't need to be done here

Reply:
I will do that. I was trying to make it generic so it can be used for other messages as well.